catroid/src/main/java/org/catrobat/catroid/ui/fragment/CategoryBricksFactory.kt

```diff
@@ -0,0 +1,777 @@
+        dataBrickList.add(ReadVariableFromFileBrick(context.getString(R.string.brick_write_variable_to_file_default_value)))
+        dataBrickList.add(AddItemToUserListBrick(BrickValues.ADD_ITEM_TO_USERLIST))
+        dataBrickList.add(DeleteItemOfUserListBrick(BrickValues.DELETE_ITEM_OF_USERLIST))
+        dataBrickList.add(ClearUserListBrick())
+        dataBrickList.add(InsertItemIntoUserListBrick(BrickValues.INSERT_ITEM_INTO_USERLIST_VALUE, BrickValues.INSERT_ITEM_INTO_USERLIST_INDEX))
+        dataBrickList.add(ReplaceItemInUserListBrick(BrickValues.REPLACE_ITEM_IN_USERLIST_VALUE, BrickValues.REPLACE_ITEM_IN_USERLIST_INDEX))
+        dataBrickList.add(WriteListOnDeviceBrick())
+        dataBrickList.add(ReadListFromDeviceBrick())
+        dataBrickList.add(StoreCSVIntoUserListBrick(BrickValues.STORE_CSV_INTO_USERLIST_COLUMN, context.getString(R.string.brick_store_csv_into_userlist_data)))
+        dataBrickList.add(WebRequestBrick(context.getString(R.string.brick_web_request_default_value)))
+        when {
+            !isBackgroundSprite -> dataBrickList.add(LookRequestBrick(BrickValues.LOOK_REQUEST))
+            projectManager.currentProject.xmlHeader.islandscapeMode() -> dataBrickList.add(BackgroundRequestBrick(BrickValues.BACKGROUND_REQUEST_LANDSCAPE))
+            else -> dataBrickList.add(BackgroundRequestBrick(BrickValues.BACKGROUND_REQUEST))
+        }
+        dataBrickList.add(AskBrick(context.getString(R.string.brick_ask_default_question)))
+        if (SettingsFragment.isAISpeechRecognitionSharedPreferenceEnabled(context)) {
+            dataBrickList.add(AskSpeechBrick(context.getString(R.string.brick_ask_speech_default_question)))
+        }
+        if (SettingsFragment.isEmroiderySharedPreferenceEnabled(context)) {
+            dataBrickList.add(WriteEmbroideryToFileBrick(context.getString(R.string.brick_default_embroidery_file)))
+        }
+        if (SettingsFragment.isAISpeechRecognitionSharedPreferenceEnabled(context)) {
+            dataBrickList.add(StartListeningBrick())
+        }
+        if (SettingsFragment.isNfcSharedPreferenceEnabled(context)) {
+            dataBrickList.add(SetNfcTagBrick(context.getString(R.string.brick_set_nfc_tag_default_value)))
+        }
+        return dataBrickList
+    }
+
+    @SuppressWarnings("ComplexMethod")
+    protected fun setupDeviceCategoryList(context: Context, isBackgroundSprite: Boolean): List<Brick> {
+        val deviceBrickList: MutableList<Brick> = ArrayList()
+        deviceBrickList.add(ResetTimerBrick())
+        deviceBrickList.add(WhenBrick())
+        deviceBrickList.add(WhenTouchDownBrick())
+        if (SettingsFragment.isNfcSharedPreferenceEnabled(context)) {
+            deviceBrickList.add(WhenNfcBrick())
+            deviceBrickList.add(SetNfcTagBrick(context.getString(R.string.brick_set_nfc_tag_default_value)))
+        }
+        deviceBrickList.add(WebRequestBrick(context.getString(R.string.brick_web_request_default_value)))
+        when {
```

Comment:
Please remove duplicated code. 
Add a function or smth similar. :)

Reply:
will be done in future refactoring